triggers/messageUndelivered.ts

```diff
@@ -0,0 +1,178 @@
+      enabled: "enabled",
+      disabled: "disabled",
+    },
+  },
+  {
+    key: "retriesEnabled",
+    required: false,
+    label: "Retries Enabled",
+    helpText:
+      "Specifies whether Whispir should perform retries in the event there is a failure accessing the callback service.",
+    choices: {
+      true: true,
+      false: false,
+    },
+  },
+  {
+    key: "email",
+    required: true,
+    label: "Email",
+    helpText:
+      "Whispir notify this email address that a callback has failed (only when retries are disabled)",
+  },
+] as const;
+
+// Step 1: Write the subscribeHook Function
+const subscribeMessageUndeliveredHook = (
+  z: ZObject,
+  bundle: InputBundle<typeof messageUndeliveredCallbackCreateInput>
+) => {
+  const inputData = transformInputData(bundle.inputData);
+
+  const data = {
+    ...inputData,
+    // the url that would be assigned automatically by Zapier
+    url: bundle.targetUrl,
+    // Hard-coding the following attributes as they are needed as part of callback resource creation by the Whispir Platform
+    // but not required as part of Zapier.
+    auth: {
+      type: "queryparam",
+      key: "not-available",
+    },
+    contentType: "json",
+    callbacks: {
+      reply: "disabled",
+      undelivered: "enabled",
+    },
+  };
+
+  const headers = {
+    "Content-Type": "application/vnd.whispir.api-callback-v1+json",
+    Accept: "application/vnd.whispir.api-callback-v1+json",
+  };
+
+  // creating a callback resource against the Whispir Platform
+  const responsePromise = z.request({
+    url: bundle.authData.host + "/callbacks",
+    method: "POST",
+    headers: addHeaders(headers, bundle),
+    body: JSON.stringify(data),
+  });
+
+  return responsePromise.then((response) => {
+    if (response.status && response.status >= 200 && response.status <= 299) {
+      return response.data;
+    } else {
+      throw new z.errors.Error(response.data);
+    }
+  });
+};
+
+// Step 2: Write the unsubscribeHook Function
+const unsubscribeMessageUndeliveredHook = (z: ZObject, bundle: Bundle) => {
+  const hookId = bundle.subscribeData?.id;
+
+  const headers = {
+    "Content-Type": "application/vnd.whispir.api-callback-v1+json",
+    Accept: "application/vnd.whispir.api-callback-v1+json",
+  };
+
+  return z
+    .request({
+      url: `${bundle.authData.host}/callbacks/${hookId}`,
+      method: "DELETE",
+      headers: addHeaders(headers, bundle),
+    })
+    .then((response) => z.JSON.parse(response.data));
+};
+
+// Step 3: Write the perform Function
+const parseMessageUndeliveredPayload = async (z: ZObject, bundle: Bundle) => {
+  const {
+    cleanedRequest: {
+      messageId,
+      from: { name, mobile, email, voice },
+      responseMessage: { content },
+    },
+  } = bundle;
+
+  const payload = {
+    id: uuidv4(),
+    messageId,
+    recipientName: name,
+    recpientMobileNumber: mobile,
+    recipientEmail: email,
+    recipientVoice: voice,
+    recpientResponseContent: content,
```

Comment:
@whutomo  Should we have `recpientResponseContent` mapped in undelivered? Or is this only fitting for messageReply received?

Reply:
Good catch, I think we should not include it here given it's for undelivered message. So, there won't be replies available for Zapier to process. I'll remove it. Thanks Sharie

Reply:
@whutomo  thanks! Before we resolve this comment, is there other payload attribute that we can add in `undelivered`? Like status or time of the status was updated?